src/profile/v100/profile.rs

```diff
@@ -190,6 +190,21 @@ impl Profile {
     }
 }
 
+impl Profile {
+    pub fn check_if_profile_json_contains_legacy_p2p_links(
+        json: impl AsRef<[u8]>,
+    ) -> bool {
+        let json = json.as_ref();
+        let result =
+            serde_json::from_slice::<ProtoProfileMaybeWithLegacyP2PLinks>(json);
+
+        match result {
```

Comment:
this can be written as:

```rust
impl Profile {
    pub fn check_if_profile_json_contains_legacy_p2p_links(
        json: impl AsRef<[u8]>,
    ) -> bool {
        let json = json.as_ref();
        serde_json::from_slice::<ProtoProfileMaybeWithLegacyP2PLinks>(json)
            .map_or_else(|_| false, |s| s.app_preferences.p2p_links.is_empty())
    }
}
```
which is a bit more clear IMO (more functional, which Sargon uses...)

Reply:
Looks better now. I only had to add a negation here: `s.app_preferences.p2p_links.is_empty()`. Thanks.

Reply:
oh right, NOT empty :)